File: JobApplicationTracker.Api/DTOs/JobApplication/JobApplicationDto.cs
namespace JobApplicationTracker.DTOs;

public class JobApplicationDto
{
    public Guid Id { get; set; }
    public string UserId { get; set; }
    public string Company { get; set; }
    public string Position { get; set; }
    public string? Note { get; set; }
    public JobListingDto JobListing { get; set; }
    public IEnumerable<JAStatusEntryDto> StatusHistory { get; set; } = [];
}

File: JobApplicationTracker.Frontend/types/JAObjects/JobApplications/JobApplicationDto.ts
import type { JAStatusEntryDto } from '../JAStatuses/JAStatusEntryDto';
import type { JobListingDto } from '../JobListing/JobListingDto';
import type { JAEventDto } from '../JAEvents/JAEventDto';

export interface JobApplicationDto {
  id: string;
  userId: string;
  company: string;
  position: string;
  note: string | null;
  jobListing: JobListingDto;
  statusHistory: JAStatusEntryDto[];
  event?: JAEventDto | null;
}